apps/src/music/blockly/MusicBlocklyWorkspace.js

```diff
@@ -330,24 +340,37 @@ export default class MusicBlocklyWorkspace {
   getLocalStorageKeyName() {
     // Save code for each block mode in a different local storage item.
     // This way, switching block modes will load appropriate user code.
-
     return 'musicLabSavedCode' + getBlockMode();
   }
 
-  loadCode() {
-    const existingCode = localStorage.getItem(this.getLocalStorageKeyName());
-    if (existingCode) {
-      const exitingCodeJson = JSON.parse(existingCode);
+  async loadCode() {
+    const projectResponse = await this.projectManager.load();
+    if (!projectResponse.ok) {
+      if (projectResponse.status === 404) {
+        // TODO: maybe rename resetCode?
+        this.resetCode();
```

Comment:
Maybe `loadDefaultCode`? Also this 404 case is expected if no sources have been saved to the project yet, right?

Reply:
good call, updated